src/lib/viewers/media/MediaBaseViewer.js

```diff
@@ -260,6 +267,17 @@ class MediaBaseViewer extends BaseViewer {
         this.wrapperEl.classList.add(CLASS_IS_VISIBLE);
     }
 
+    /**
+     * Refresh the access token
+     *
+     * @private
+     * @return {Promise<Object>}
+     */
+    refreshToken = async () => {
+        const tokenMap = await getTokens(this.options.file.id, this.options.tokenGenerator);
```

Comment:
Could we instead have `Preview.js` be the only place to set/generate tokens? Perhaps when we `createViewerOptions` we could pass in an callback to `refreshToken`